### DISCUSSION TITLE
[X40 Ultra] "Mop Dock Auto Repair" quirk functionality

### DISCUSSION THREAD

**Discussion by abueschel** [Hypfer/Valetudo#2101] [Q&A / Support]
### Are you already using Valetudo?

Yes, for a long time

### Your questions

Hi,

I noticed my Dreame X40 Ultra started to have an issue keeping the mop pads wet long enough during cleanup runs, so I tried using the firmware quirk "Mop Dock Auto Repair", as the description fits my problem:

```
If the mops don't stay wet during cleaning, there might be air in the system,preventing the water tank in the robot from being filled. In that case, try running this a few times.
```

My question is: How long is this task supposed to run? When I trigger the task, the robot audibly starts doing something (pumping I guess). I measured the time once, it took over one hour before I cancelled it by rebooting the robot. The robot also does not consume any water from the tank during this time, which it should be to get the air out of the system I guess? I also did not notice any change in the behaviour of the mop wetness after triggering the quirk.

```
Valetudo 2024.06.2
Firmware 1252
```

The robot is fairly new, I used it for a couple weeks without any cloud to verify it's not a defective unit and it's running Valetudo for a bit under a month now. I had a L10 Pro running Valetudo before for about two years.

**Comment by TheUltimateC0der** [comment]
I am also interested in some info here. My robot also has the problem that the mop pads get dry rather quick. 

When running this via the webui, it seems to be an endless task that cannot be stopped other than rebooting the robot. 

It also seems odd to me that there doesn't seem to be any water involved since I marked the water level on the tank of the station. 

Can anyone explain what the robot actually does ?

**Comment by K0n24d** [comment]
Hello,

I'm running into the same issue and observations. Did you find any answer? Or a way to resolve the pads getting dry?

Thanks

**Reply by K0n24d** [reply]
Just if anybody comes here with the same issue regarding the mops not staying wet. Mine was still under warranty so I RMA'd it. They replaced the pump and tubings inside the robot.

On a side note I flashed it with a firmware without the DNS hack before sending it to them and did not get any remarks from them regarding the firmware

**Comment by bismuth89** [comment]
Hi, anyone has figured out an answer for this? I got a L40 ultra with exact same problem: mop dried out quickly during cleaning, ran the quirks to fix it but it is never ending until I shut down the robot and the problem is not fixed. my robot is on valetudo and out of waranty. Woule be really greatful if someone could help

**Reply by TheUltimateC0der** [reply]
From what I have learned: The quirk will never finish. The problem (at least for me) was that the lines were full of lime. You can either try to fix this by poking in the holes for the mops with a toothpick to loosen the lime, or partly disassemble the robot to try and fix it.

**Reply by bismuth89** [reply]
Thank you for replying! I found the solution today after digging in reddit and I think it is similar with what you said. For my case the inner tank got water from dock but it is not dispensing to the mops during cleaning, so the trick of pumping water using syringe through the clean water tank hole in the dock doesnt work. Finally fixed it by using a syringe with needle (cut short the needle) and pump water directly to the robot tank through the hole in the back until water starts coming out through the hole of 1 mop. Then pump water through that mop hole till it comes out the other holes (or if not you can pump the other holes as long as it feels like all the holes got water through them). Seems like something was stuck but now it is working!
